.github/workflows/generate-summary.yaml

```diff
@@ -187,11 +191,70 @@ jobs:
           unzip ./temp/*report.log.zip -d ./current_logs || true
           ls current_logs
 
+      - name: Build Log Comparison Setup
+        id: build-log-setup
+        run: |
+          echo "previous_build_logs_path=$(readlink -f previous_build_logs)" >> $GITHUB_OUTPUT
+          echo "previous_build_log_zips_path=$(readlink -f previous_build_log_zips)" >> $GITHUB_OUTPUT
+          echo "current_build_logs_path=$(readlink -f current_build_logs)" >> $GITHUB_OUTPUT
+          echo "current_build_log_zips_path=$(readlink -f current_build_log_zips)" >> $GITHUB_OUTPUT
+          mkdir new_build_warnings
+          echo "new_build_warnings_path=$(readlink -f ./new_build_warnings/new_build_warnings.log)" >> $GITHUB_OUTPUT
+
       - name: Download artifacts
+        id: download-previous-artifacts
         run: |
           pip install pygithub requests
-          python ./scripts/download_artifacts.py -hash ${{ inputs.gcchash }} -repo patrick-rivos/gcc-postcommit-ci -token ${{ secrets.GITHUB_TOKEN }} -prefix "${{ inputs.prefix }}"
+          python ./scripts/download_artifacts.py -hash ${{ inputs.gcchash }} -repo patrick-rivos/gcc-postcommit-ci -token ${{ secrets.GITHUB_TOKEN }} -prefix "${{ inputs.prefix }}" -build-logs -build-logs-dir ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
           ls previous_logs
+          if [ -d "${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}" ]; then
+            ls ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
+          fi
+
+      - name: Unzip previous build log artifacts
+        id: unzip-previous-build-log-artifacts
+        uses: ./.github/actions/common/unzip-all-zips
+        with:
+          input-dir: ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
+          output-dir: ${{ steps.build-log-setup.outputs.previous_build_logs_path }}
+          include-pattern: "*-stderr.log"
+
+      - name: Download current build log artifacts
+        id: download-current-build-log-artifacts
+        if: github.event_name != 'workflow_dispatch'
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "*-build-log"
+          merge-multiple: true
+          path: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+
+      - name: Download current build log artifacts manual
+        id: download-current-build-log-artifacts-manual
+        if: (github.event_name == 'workflow_dispatch') && (inputs.build_log_current_run_id != '')
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "*-build-log"
+          merge-multiple: true
+          path: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ inputs.build_log_current_run_id }}
```

Comment:
I'm a bit confused. what happens if we use workflow dispatch without specifying the run id? Neither of these two steps would be hit and nothing would be downloaded. Is that intended?

Reply:
Yes if the run-id is not specified, build log warning comparison is disregarded.